### ISSUE TITLE
Query Predicate: enhancement request: IS [NOT] NULL 

### ISSUE THREAD

**Issue by ScottNicol** [hazelcast/hazelcast#6189]
IS [NOT] NULL would be useful for null checking values. If this is not possible, then a NULL keyword that resolves to the value null would be helpful, so you could write x = NULL or x != NULL as a substitute for IS NULL and IS NOT NULL.

Thanks for fixing issue 6186. I was just about to report this, because without it x != NULL doesn't provide a useful result.

**Comment by jerrinot** [comment]
@ScottNicol: Thanks for you suggestions. 
They seem sensible to me; I'll have a look what we can do about it.

**Comment by jerrinot** [comment]
@ScottNicol: I believe it's actually working even now. These tests are passing for me:
```java
    @Test
    public void testEqualsNull() {
        HazelcastInstance instance = createHazelcastInstance();
        String mapName = randomMapName();
        IMap<Integer, SampleObjects.ValueType> map = instance.getMap(mapName);
        map.put(1, new ValueType("notNull"));
        map.put(2, new ValueType(null));
        Predicate predicate = new SqlPredicate("typeName = null");
        Set<Map.Entry<Integer, ValueType>> entries = map.entrySet(predicate);
        assertThat(entries, hasSize(1));
        assertEquals(2, (long)entries.iterator().next().getKey());
    }
```

```java
    @Test
    public void testNotEqualsNull() {
        HazelcastInstance instance = createHazelcastInstance();
        String mapName = randomMapName();
        IMap<Integer, SampleObjects.ValueType> map = instance.getMap(mapName);
        map.put(1, new ValueType("notNull"));
        map.put(2, new ValueType(null));
        Predicate predicate = new SqlPredicate("typeName != null");
        Set<Map.Entry<Integer, ValueType>> entries = map.entrySet(predicate);
        assertThat(entries, hasSize(1));
        assertEquals(1, (long)entries.iterator().next().getKey());
    }
```

Is this what you are asking for?

**Comment by ScottNicol** [comment]
Write a SQL query "A IS NULL".

The IS predicate is not supported.

A non-standard workaround is "A = NULL", however there is no NULL constant.  You can use "A = aValueThatIsNull".

Then there's "A IS NOT NULL". An obvious workaround is "A != aValueThatIsNull", but until yesterday (when issue 6186 was fixed) that didn't work, so the workaround was "! ( A = aValueThatIsNull )"

**Comment by jerrinot** [comment]
@ScottNicol: We could add the "is null / is not null" quite easily. It could be just an alias producing the same nodes as  "= null" or " != null".

However I'm not sure what you mean by "there is no NULL constant". A parser inside SQLPredicate recognizes the "null" token. 

You can use either this: `Predicate p = Predicates.equals("attribute", null);` or if you prefer the SQL-like version you can use `Predicate p = new SQLPredicate("attribute = null");` Both versions are equivalent.  

Did I miss anything?

**Comment by ScottNicol** [comment]
You're right about NULL, sorry about that. I probably got confused about that when I was checking out the != behavior. I just tried it, NULL works.

So IS/IS NOT could be aliased to =/!= and that would fix things. In SQL the only thing that can follow IS [NOT] is NULL, but I don't think it's too import, if it complicates the parsing.

Thanks for your response.

**Comment by jerrinot** [comment]
I'm glad it's clear now. 
I'll keep this one in a backlog. Thanks for your feedback!
